src/x11/x.c

```diff
@@ -908,9 +910,17 @@ gboolean x_mainloop_fd_dispatch(GSource *source, GSourceFunc callback, gpointer
                         break;
                 case FocusIn:
                 case FocusOut:
-                case PropertyNotify:
                         wake_up();
                         break;
+                case PropertyNotify:
+                        scr = get_active_screen();
+                        /* Ignore PropertyNotify, when we're still on the
+                         * same screen. PropertyNotify is only neccessary
+                         * to detect a focus change to another screen
+                         */
+                        if(scr->scr != xctx.cur_screen)
```

Comment:
You can limit when this runs even more by checking that `settings.f_mode != FOLLOW_NONE`

Reply:
Handled now.